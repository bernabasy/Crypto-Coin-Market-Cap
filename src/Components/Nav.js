import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (
  <nav className="nav bg-warning">
    <ul className="d-flex justify-content-between w-100 my-2 mx-0">
      <li className="fonts d-flex gap-2 list-unstyled">
        <NavLink
          className="text-decoration-none"
          to="/"
        >
          <h3 className="fonts"> â®Œ</h3>
        </NavLink>
        <h5 className="pl-3 mt-1 fonts pad">Crypto Coin Market Cap</h5>
      </li>
      <li className="d-flex list-unstyled px-0">
        <h1>ðŸŽ™</h1>
        <h1 className="px-2 font-weight-bold">â˜¼</h1>
      </li>
    </ul>
  </nav>
);

export default Nav;
