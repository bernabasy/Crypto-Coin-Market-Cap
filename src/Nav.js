import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (
  <nav>
    <div>Financial Modeling</div>
    <ul>
      <li>
        <NavLink
          to="/"
        >
          homepage
        </NavLink>
      </li>
      <li>
        <NavLink to="/details">
          Details
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Nav;
