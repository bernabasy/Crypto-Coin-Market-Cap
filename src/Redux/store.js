import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './Modeling';

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
});
export default store;
