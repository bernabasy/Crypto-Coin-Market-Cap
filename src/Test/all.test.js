import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Nav from '../Components/Nav';
import Models from '../Components/renderCrypto';
import Crypto from '../Components/createCrypto';

const crypto = { FULLName: 'bitcoin', img: 'new' };
jest.mock('axios');

describe('Nav component', () => {
  it('renders Nav component', () => {
    const r = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(r).toMatchSnapshot();
  });
  it('renders Models component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Models />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
  it('renders Crypto component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Crypto crypto={crypto} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});
