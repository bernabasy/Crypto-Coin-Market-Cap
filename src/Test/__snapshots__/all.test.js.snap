// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Crypto component 1`] = `
<div
  className="rounded-4 w-100 h-100 border border-dark d-flex justify-content-center"
>
  <div
    className="w-100"
  >
    <p
      className=" p-2 font text-light"
    >
       
      <span>
        Name:
      </span>
    </p>
    <div
      className="d-flex m-3 justify-content-center"
    >
      <img
        alt="symbol"
        className="images w-75"
        src="https://www.cryptocompare.comnew "
      />
    </div>
    <p />
  </div>
</div>
`;

exports[`Nav component renders Models component 1`] = `
Array [
  <form
    className="w-100 mb-5 bg-secondary nav"
  >
    <input
      aria-label="search"
      className="input w-100 h-2"
      onChange={[Function]}
      placeholder=".....search you cryptos"
      type="text"
      value=""
    />
  </form>,
  <div
    className="bg-primary text-white"
  >
    <div
      className="grid1"
    />
  </div>,
]
`;

exports[`Nav component renders Nav component 1`] = `
<nav
  className="nav bg-warning"
>
  <ul
    className="d-flex justify-content-between w-100 my-2 mx-0"
  >
    <li
      className="fonts d-flex gap-2 list-unstyled"
    >
      <a
        aria-current="page"
        className="text-decoration-none active"
        href="/"
        onClick={[Function]}
      >
        <h3
          className="fonts"
        >
           â®Œ
        </h3>
      </a>
      <h5
        className="pl-3 mt-1 fonts pad"
      >
        Crypto Coin Market Cap
      </h5>
    </li>
    <li
      className="d-flex list-unstyled px-0"
    >
      <h1>
        ðŸŽ™
      </h1>
      <h1
        className="px-2 font-weight-bold"
      >
        â˜¼
      </h1>
    </li>
  </ul>
</nav>
`;
